packages/lib/test/src/artifacts/ZWeb3.test.js

```diff
@@ -189,6 +189,34 @@ contract('ZWeb3', accounts => {
           })
         })
       })
+
+      describe('checksum address', function () {
+        context('when address has uppercase hexa letters', function () {
+          context('when address is not well checksummed', function () {
+            it('fails', function () {
+              try {
+                const address = '0x82154A1cbcAb0461D4C12ee9Cb890244eead6d9F';
+                const checksummed = ZWeb3.toChecksumAddress(address);
+              } catch(error) {
+                error.message.should.match(/is not a valid Ethereum address or it has not been checksummed correctly/);
+              }
```

Comment:
Note that this test does not actually test that it fails! If the call does not throw, the catch is never executed, and the test succeeds. Look for the `expects(fn).to.throw()` method [in chai](https://www.chaijs.com/api/bdd/#method_throw) to implement this.

Reply:
Fixed!